sklearn/svm/_liblinear.pyx

```diff
@@ -14,36 +14,86 @@ include "_liblinear.pxi"
 cnp.import_array()
 
 
-def train_wrap(X, cnp.ndarray[cnp.float64_t, ndim=1, mode='c'] Y,
-               bint is_sparse, int solver_type, double eps, double bias,
-               double C, cnp.ndarray[cnp.float64_t, ndim=1] class_weight,
-               int max_iter, unsigned random_seed, double epsilon,
-               cnp.ndarray[cnp.float64_t, ndim=1, mode='c'] sample_weight):
+def train_wrap(
+    object X,
+    cnp.float64_t[::1] Y,
+    bint is_sparse,
+    int solver_type,
+    double eps,
+    double bias,
+    double C,
+    cnp.float64_t[:] class_weight,
+    int max_iter,
+    unsigned random_seed,
+    double epsilon,
+    cnp.float64_t[::1] sample_weight
+):
     cdef parameter *param
     cdef problem *problem
     cdef model *model
     cdef char_const_ptr error_msg
     cdef int len_w
+    cdef bint x_has_type_float64 = X.dtype == np.float64
```

Comment:
Even if every variable in Python must be snake cased, in scikit-learn we do capitalize `X` and related variables. In this case, we would have:

```suggestion
    cdef bint X_has_type_float64 = X.dtype == np.float64
```

This suggestions also applies to similar variable hereinafter.